packages/cli/src/rpc/explain/explain.ts

```diff
@@ -150,6 +150,20 @@ async function explain(
       searchStatusByUserMessageId.set(userMessageId, status);
       first() && resolve({ userMessageId, threadId });
     });
+
+    // If the request completes here, consider it an error. This would mean that the
+    // remote server terminated our connection early.
+    explain.on(
+      'complete',
+      () =>
+        first() &&
+        reject(
+          RpcError.fromException(
+            new Error(status.explanation?.join('') || 'The response completed unexpectedly')
+          )
+        )
+    );
```

Comment:
I could see a case made for this being it's own error message or something similar, though we're not doing anything more with these messages where there's an explicit need to differentiate.

From the servers perspective:
1. A user connects
2. The server determines the user cannot proceed
3. A response is written to the client as raw tokens
4. An `end` message is sent and the connection is terminated